client/components/ProductOverview/ProductOverview.jsx

```diff
@@ -23,23 +47,55 @@ const ProductOverview = ({ productId }) => {
     setImageIndex(index);
   };
 
-  return (
-    <div>
-      <div className="productOverviewContainer" data-testid="productOverView">
-        <ImageGallery currentStyle={currentStyle} id="imageGallery" imageIndex={imageIndex} updateImageIndex={updateImageIndex} />
-        <div className="productInfo">
-          <div>Stars</div>
-          <div>Category</div>
-          <div>Name</div>
+  const updateCurrentStyle = (index) => {
+    setStyleIndex(index);
+    setCurrentStyle(allStyles[index]);
+  };
+
+  if (currentStyle) {
+    return (
+      <div>
```

Comment:
This is a really big return statement with a lot going on. I'm wondering if there's any way to break apart some of this into smaller components? Social media could be its own component etc. Also, you can use a ternary rather than the **if** statement like 
```
return !currentStyle ? <div>Loading</div> : <div>All the long component stuff code</div>;

Reply:
honestly, this condition return is a workaround. On mount the ProductOverview component needs to complete an axios call to get the currentstyle. Since that is async, the component will try to render before it has all of the information it needs and then it breaks. That's why I set an if statement to return a basic "Loading Product Info" div while the axios request is finishing. Not sure if this is considered hacky. Would like your thoughts.